src/main/python/hydra/lib/runtestbase.py

```diff
@@ -221,6 +222,74 @@ def create_hydra_app(self, name, app_path, app_args, cpus, mem, ports=None, cons
         self.refresh_app_info(name)
         return r
 
+    def create_hydra_app_group(self, name, app_path, app_args, cpus, mem, apps_in_group,
+                               total_apps, ports=None, constraints=None):
+        """
+        Create an application in groups
+        @args:
+        name:                                      name of the app
+        app_path:                                  path to launch app from
+        app_args:                                  app args
+        cpus:                                      required cpus
+        mem:                                       required  mem
+        apps_in_group:                             number of apps to group together
+        total_apps:                                total apps to launch
+        ports:                                     port list (optional)
+        constaints:                                constaints optional
+        """
```

Comment:
I dont think the group construct will be used to create application, instead created application will be added to groups. 
Doing it this way leaves it open to some one calling scale and causing all your data structures to get messed up. 

It would be better to re-use what's already there and add the group layer on top of it, so just add a data structure to track the application-id's or the combination of "task_id" + "port"